src/components/AR/AR-session.ts

```diff
@@ -0,0 +1,95 @@
+import { WonderlandEngine } from '@wonderlandengine/api';
+import { ARProvider } from './AR-provider.js';
+
+/**
+ * ARSession - master control for the AR session.
+ * - loads dependencies (aka providers)
+ * - handles global callbacks when AR session is started, ended
+ * - can end any running AR session.
+ * - renders AR button when the scene is loaded
+ */
+abstract class ARSession {
+    // tracking provider is basically a lib which has some tracking capabilities, so device native webXR, 8th Wall, mind-ar-js, etc
+    private static _trackingProviders: Array<ARProvider> = [];
+
+    // current running provider
+    private static _currentTrackingProvider: ARProvider | null = null;
+
+    public static readonly onARSessionReady: Array<() => void> = [];
+
+    public static readonly onSessionStarted: Array<(trackingProvider: ARProvider) => void> =
+        [];
+    public static readonly onSessionEnded: Array<(trackingProvider: ARProvider) => void> =
+        [];
+
+    private static _sceneHasLoaded = false;
+    private static _arSessionIsReady = false;
+
+    public static get arSessionReady() {
+        return this._arSessionIsReady;
+    }
+
+    /**
+     * Registers tracking provider. Makes sure it is loaded
+     * and hooks into providers onSessionStarted, onSessionLoaded events.
+     */
+    public static async registerTrackingProvider(engine: WonderlandEngine, provider: ARProvider) {
+        if(!engine.onSceneLoaded.includes(this.onWLSceneLoaded)) {
+            engine.onSceneLoaded.push(this.onWLSceneLoaded);
+        }
+
+        if (this._trackingProviders.includes(provider)) {
+            return;
+        }
+        
+        this._trackingProviders.push(provider);
+        provider.engine = engine;
```

Comment:
Is that going to create issues with 2 engines?

Scenario:
* I have 2 engines
* I call `registerTrackingProvider` on the first one
* I call `registerTrackingProvider` on the second one

What happens? Having 2 engines is now a supported feature, like you could have 2 `WebGLRenderer` with Three.js